src/JuMP.jl

```diff
@@ -169,23 +162,29 @@ mutable struct Model <: AbstractModel
     # dictionary keyed on an extension-specific symbol
     ext::Dict{Symbol,Any}
     # Default constructor
-    function Model(; simplify_nonlinear_expressions::Bool=false)
+    function Model(; mode::ModelMode=Automatic, solver=nothing, simplify_nonlinear_expressions::Bool=false)
         # TODO need to support MPB also
         m = new()
         # TODO make pretty
-        m.instance = JuMPInstance{Float64}()
         m.variabletolowerbound = Dict{MOIVAR,MOILB}()
         m.variabletoupperbound = Dict{MOIVAR,MOIUB}()
         m.variabletofix = Dict{MOIVAR,MOIFIX}()
         m.variabletointegrality = Dict{MOIVAR,MOIINT}()
         m.variabletozeroone = Dict{MOIVAR,MOIBIN}()
-        m.variablestart = VariableToValueMap{Float64}(m)
-        m.variableresult = VariableToValueMap{Float64}(m)
         m.customnames = Variable[]
         m.objbound = 0.0
         m.objval = 0.0
-        m.solverinstance = nothing
-        m.solverinstanceattached = false
+        if mode == Automatic
+            @assert solver === nothing
```

Comment:
Why not do `resetsolver!` in case it is not `nothing` ?

Reply:
Done